### ISSUE TITLE
August to-do list for Sean (what should be on it?)

### ISSUE THREAD

**Issue by sean-fitzpatrick** [APEXCalculus/APEXCalculusPTX#189]
Next month I'm hoping to put in some time doing some larger-scale tasks with APEX.
(Right now I'm travelling, and working from a laptop limits productivity a bit -- I miss my external monitors!)

What should we prioritize?

First should be the images where captions are cut off, and especially those where the function is not described in the exercise itself.
Second: adding tasks to exercises? This is best practice but I'm not sure how important it is.
Third: There are a few issues related to technical issues with definitions and theorems; I can update that content if there's agreement on how to change it.

The issue of images lacking descriptions remains. I doubt I can tackle that in a week or two, but I could make a start.

I'll also need to rebuild PDF, which will once again involve making sure that margin images are aligned correctly.
I can save a bit of time by using last year's values for all the vertical adjustments, either using git or a good diff tool.

Long term I'd still like to find a solution that doesn't require me to build the PDF, manually edit each margin figure to add vertical shifts based on a visual inspection of the PDF, rebuild, adjust again, and repeat until it looks good.

The best idea I've come up with is to use our custom XSL to add attributes to figures that store this information, but once we do that, we're sort of moving away from proper PreTeXt.

**Comment by sean-fitzpatrick** [comment]
The trouble we have now is that the `.tex` file is generated from PreTeXt source, and at the moment, there is no way to store those values in the PreTeXt source. So when you rebuild the LaTeX source, none of the shift values are there, and you have to add them in.

Two solutions have worked for me in the past:

1. Copy new LaTeX source to a directory tracked by git. Make all the necessary changes as a git commit. When new source is built, apply that git commit to it to add in the corresponding changes.
2. Use a diff tool like Meld to show old and new source side-by-side. Lines where shifts have been added will be highlighted, and you just need to click a button to copy the change over.

The first option is faster, but prone to errors if the new version of the LaTeX source has diverged too far from the previous.

I couldn't find a way to efficiently transfer work from one version of the book to another. (I have two versions of the book that present the material in different orders, so sections have moved around, making a git comparison impossible.)
If shifts were tracked in PreTeXt source this would be a non-issue but I don't know how to do this without breaking the Laws of PreTeXt.

**Comment by sean-fitzpatrick** [comment]
@Alex-Jordan is currently working on replacing the older 'PG' graphs with TikZ in WeBWorK problems.

We had discussed the problem of finding a more robust way to add labels to TikZ graphs so that they don't get accidentally cropped. (I think you pointed out this issue.)
In some of those exercises, you had also pointed out that in some cases, the definition of the function occurred only in the image, and not in the problem text, which is problematic both for accessibility and because problems like cropping can occur.

**Comment by Alex-Jordan** [comment]
I moved chapter 1's WW images to tikz, but that is definitely a demonstration for someone else and only maybe a sign that I myself can continue :)

I do still need to do one where the tikz image is in an exercisegroup introduction. Just to demo that this works as well.

These recent changes require a WW 2.17 server, with 2.17 being officially released just two days ago. Before that, there was an awkwardness to how it worked that we wanted to go away. So we did not document it. But now it's mostly documented here:

Paragraph 
https://pretextbook.org/doc/guide/html/webwork-source.html#p-992

One thing to be aware while plotting functions is:
https://webwork.maa.org/moodle/mod/forum/discuss.php?d=8084

And it helps to really know tikz/pgfplots syntax well. Currently, if you have broken syntax, the error messages are so far upstream that PTX processing does not reveal what exactly was wrong. I have a process I go through where I copy the human-readable version of a bad problem into the host course and run it there. It's too cumbersome to document though. Long term, we will have better error reporting.

All of this works with pretext/pretext/pretext. I am unsure if it has made its way to the CLI yet.

**Comment by Alex-Jordan** [comment]
The issue about labels on tikz graphs is here:
https://github.com/APEXCalculus/APEXCalculusPTX/issues/161

And the section mentioned is here:
https://opentext.uleth.ca/apex-video/sec_disk.html

At least for the graphs in that section, it's not sufficient (for accessibility) to put a label of the curve outside the graph. It is also important to describe the entire boundary of the shape being rotated. So what would normally be needed is to include all that in the alt text (`description`). It may not fit, since practically, alt text is capped at 140 characters (or something of that nature). We are looking into good ways to present a long description of an image. But I'm not sure what to do right now for accessibility.

But my point is that I think it is OK to focus now on fixing the visual presentation of the curve labels. Accessibility will be handled in alt text or a long description.

I'm a fan of the legend approach, but it can be labor intensive.

**Comment by sean-fitzpatrick** [comment]
Right. Legends could work, but then I think we'd want to do it throughout the book.

And we have to figure out how to get the styling to look like we want it to. (Which also requires us to agree on how it should look.)

**Comment by Alex-Jordan** [comment]
> Which also requires us to agree on how it should look.

If the styling is controlled globally, we can experiment with different styles easily. I think this won't be so hard, except for occasionally needing to position a legend differently from a default so it doesn't cover something important.

**Comment by sean-fitzpatrick** [comment]
I am going to put in some work today on the graphs in the disk/washer and shells sections.
But it will be a temporary fix: mention the function in the question text, and try to adjust the image so the label is not cut off.

A complete overhaul of labels will have to come later: the fall semester is getting too close!

I could also split off video solutions into separate proofs, if there is any appetite for that.

**Comment by sean-fitzpatrick** [comment]
There are 5 open (and relatively small) pull requests that I think should address issues #152, #155, #161, #166, and #173.

Once those are merged, I think I'd like to do a freeze so I can build PDF for the fall.

Is there anything else that should be tackled before then? Moving all the examples to use `<task>` is too much work for right now; so is adding image descriptions, and doing more with webwork-TikZ.

We discussed maybe putting video solutions in their own knowl. I could make that change pretty quickly, if others are in agreement.

**Comment by sean-fitzpatrick** [comment]
Oh, area between curves has a bunch of exercises with the same issues as #161. So I should try to tackle that before doing anything else.

**Comment by sean-fitzpatrick** [comment]
My to-do is to-done, for now. (Aside from taking care of remaining PR merges, etc.)

**Comment by Alex-Jordan** [comment]
Thanks Sean, I wish I could have been more help this summer. But you got a lot done!

**Comment by sean-fitzpatrick** [comment]
Turns out the secret is to get a mild case of covid. I tested positive yesterday morning, so I've been isolating in my basement since, with nothing to do but work at my computer.

BTW I'm quite happy with how the video solutions change worked out. (The XSL in particular.)
The only thing I'd still like to change are the titles:

Right now in HTML, we get:

> Solution 1.  Video Solution     > Solution 2.

I'd rather just have

> Video Solution    > Text Solution

but I think that requires further template overrides.
